src/main/java/io/vertx/ext/amqp/AmqpConnection.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2018-2019 The original author or authors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ *        The Eclipse Public License is available at
+ *        http://www.eclipse.org/legal/epl-v10.html
+ *
+ *        The Apache License v2.0 is available at
+ *        http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.amqp;
+
+import io.vertx.codegen.annotations.Fluent;
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+
+
+/**
+ * Once connected to the broker or router, you get a connection. This connection is automatically opened.
+ */
+@VertxGen
+public interface AmqpConnection {
+
+  /**
+   * Registers a handler called on disconnection.
+   *
+   * @param endHandler the end handler.
+   */
+  @Fluent
+  AmqpConnection endHandler(Handler<Void> endHandler);
```

Comment:
use `closeHandler` instead, that's how it is usually called when you want to be aware of the close notification

Reply:
There is a closeHandler also. I thought both were usually defined, with endHandler called for 'unexpected' closure?

Reply:
`endHandler` is usually used when there is a `ReadStream`, what's the purpose of this ?

Reply:
As implemented, it fires if the connection remotely closes or disconnects without you asking for it to.